src/filelogger.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  existsSync,
+  accessSync,
+  constants as fsConstants,
+  writeFileSync,
+  createWriteStream,
+  WriteStream,
+  mkdirSync
+} from 'fs'
+import { dirname as directoryname } from 'path'
+
+export class FileLogger {
+  private readonly filename: string
+  private readonly dirname: string
+  private writableStream: WriteStream
+  constructor(filename: string) {
+    this.dirname = directoryname(filename)
+    this.filename = filename
+    this._stat()
+    this._createWritableStream()
+  }
+
+  private _stat() {
+    //check if file exists
+    if (existsSync(this.filename)) {
```

Comment:
`existsSync` is deprecated, you should use `fs.access` API with a special code, I think it's `F_OK`